debezium-embedded/src/test/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotTest.java

```diff
@@ -470,6 +493,40 @@ public void snapshotPreceededBySchemaChange() throws Exception {
         }
     }
 
+    @Test
+    public void testPauseDuringSnapshot() throws Exception {
+        populateTable();
+        final Configuration config = config().build();
+        startAndConsumeTillEnd(connectorClass(), config);
+        waitForConnectorToStart();
+
+        waitForAvailableRecords(1, TimeUnit.SECONDS);
+        // there shouldn't be any snapshot records
+        assertNoRecordsToConsume();
+
+        sendAdHocSnapshotSignal();
+        // Sleep a bit to stream some events.
+        Thread.sleep(200);
```

Comment:
I am not sure if this is reliable, would not it be better to either listen on JMX metrics or catching some incoming messages to make sure the few snapshot events have arrived?

Reply:
Yes, this is not a very good way how to do it. Thanks for suggestions, will take a look on it.

Reply:
done